### PR TITLE
Remove unused argument in sync_index

### PR DESCRIPTION
**Pull request by pascutto**

### CODE REVIEW

File: src/index.ml [hunk 1]
```diff
@@ -237,17 +237,17 @@ struct
         else if offset < h.offset then sync_log_entries ~min:offset log
         else if offset > h.offset then assert false
 
-  let sync_index ~generation t =
-    may (fun (i : index) -> IO.close i.io) t.index;
+  let sync_index t =
+    Option.iter (fun (i : index) -> IO.close i.io) t.index;
```
Comment (CraigFe) [inline src/index.ml:241]: Can do likewise for all occurrences of `may` and then get rid of it.

File: src/index.ml [hunk 2]
```diff
@@ -237,17 +237,17 @@ struct
...
     let index_path = Layout.data ~root:t.root in
-    if not (IO.exists index_path) then t.index <- None
-    else
+    if IO.exists index_path then
       let io =
-        IO.v ~fresh:false ~readonly:true ~generation ~fan_size:0L index_path
+        IO.v ~fresh:false ~readonly:true ~generation:0L ~fan_size:0L index_path
```
Comment (CraigFe) [inline src/index.ml:245]: Might warrant a comment saying that this argument isn't used in this combination.

**Comment by pascutto** [comment]
There are multiple places where your comments apply; if you don't mind waiting a bit, there's a subsequent PR coming soon that fixes them all.

**Comment by pascutto** [comment]
Fixed in #276

**Pull request merged**